src/common/ceph_argparse.cc

```diff
@@ -98,15 +93,7 @@ void env_to_vec(std::vector<const char*>& args, const char *name)
   if (!name)
     name = "CEPH_ARGS";
 
-  bool dashdash = false;
-  std::vector<const char*> options;
-  std::vector<const char*> arguments;
-  if (split_dashdash(args, options, arguments))
-    dashdash = true;
-
-  std::vector<const char*> env_options;
-  std::vector<const char*> env_arguments;
-  std::vector<const char*> env;
+  auto [options, arguments] = split_dashdash(args);
```

Comment:
We need to backport to Mimic -- was C++17 supported?

Reply:
yeah. https://github.com/ceph/ceph/blob/mimic/src/CMakeLists.txt#L149